### PR TITLE
Default kube-proxy to the old behavior for proxier sync.

### LINKED ISSUE #36281: Investigate kube-proxy sluggishness
We observed weird sluggishness in some kubeproxies while debugging https://github.com/kubernetes/kubernetes/issues/36266

### PR DESCRIPTION
**Pull request by MrHohn**
Fix #36281.

This PR defaults `minSyncPeriod` to 0 and makes kube-proxy fall back to the old behavior to fix the immediate problem #36266.

@bprashanth

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36332)

### CODE REVIEW

**Comment by MrHohn** [comment]
@k8s-bot cvm gke e2e test this

File: pkg/proxy/iptables/proxier.go [hunk 1]
```diff
@@ -252,16 +252,23 @@ func NewProxier(ipt utiliptables.Interface, sysctl utilsysctl.Interface, exec ut
 
 	go healthcheck.Run()
 
-	syncsPerSecond := float32(time.Second) / float32(minSyncPeriod)
+	var throttle flowcontrol.RateLimiter
+	if minSyncPeriod == 0 {
+		// Defaulting back to not limit sync rate.
+		throttle = nil
```
Comment (bprashanth) [inline pkg/proxy/iptables/proxier.go]: nit, you can just assign the actual value if minSync != 0

**Review by bprashanth — COMMENTED** [review]
LGTM, the nit is not significant. I'd like this to go in because there's quite a bit of flake, and if it isn't going to fix it we need to debug more. Thanks for the pr.

**Comment by MrHohn** [comment]
Saw the lgtm is not applied yet, so I re-pushed for the nit.

**Comment by bprashanth** [comment]
Oh what, i thought i lgtm'd this

**Comment by timothysc** [comment]
@bprashanth you're essentially incurring a non-trivial load on a number of environments to sustain tests that want a faster response.  b/c the watch was per-resource versions updates would batch appropriately.

**Comment by bprashanth** [comment]
there is a lot that goes on inbetween a watch (i.e your reflector) and the syncproxyrules (i.e where you had the accept() rate limit)

**Comment by bprashanth** [comment]
I mean, they *won't* batch appropriately and you will see janky performance (o(10s) as the test showed us)

**Comment by thockin** [comment]
This partial-revert seems worse than a total revert, to me.  I didn't follow all of the triage - is this demonstrably broken and I am just not seeing why?  Or is it just bad for tests?

What is the plan for a real fix here?  If it is broken and we don't have a plan, we should revert the whole thing..

**Comment by bprashanth** [comment]
the real fix is to avoid flooding syncProxyRules with no-op updates vis smarter batching. the flag still makes sense if we do that, io ratelimits are usually a good thing. Currently we rate limit, but the queue of no-op updates just builds up causing jitter in service create/update/nodeport opening etc at the order of 10s (and of course submit queue blockage because all the Services tests flake a lot), which presumably is exactly what we were trying to alleviate with a rate limit.

**Comment by thockin** [comment]
Yeah, we shouldn't be queueing.  We should be accumulating and syncing all
at once.  TCP_CORK style.

**Comment by timothysc** [comment]
I can just switch to TryAccept which was one of the versions.

**Comment by bprashanth** [comment]
TryAccept will drop, which is equally bad, unless you're also going to remember what didn't go through and apply in the successful accept

**Comment by timothysc** [comment]
it will non-block, but the data structure will be updated, and we can reset the synch timer on failed tryaccept.

**Comment by thockin** [comment]
We're going to sync from the cache, right, so as long as the cache
accumulates deltas and we have a timer to resync at the next opportunity we
should be OK.  Under load, we'll settle into a pattern of one sync every
couple seconds.

On Wed, Nov 30, 2016 at 8:48 AM, Timothy St. Clair <notifications@github.com

**Comment by bprashanth** [comment]
I just don't want to rely on the value of IPTablesSyncPeriod being something small to actually pick up and apply Service ip/port updates, but yeah, a periodic sync will eventually fix everything

**Comment by thockin** [comment]
The existing periodic sync is OK to be large because it assumes no
cause-based syncs.  If we have a lot of cause-based syncs (from real
changes) and a rate-limit, it effectively becomes a periodic sync at a
higher rate.  If changes slow down or stop coming, the resync becomes less
periodic and more on-demand.  That's what I envisioned with the PR in the
first place.

**Pull request merged**